KarhooUISDK/Screens/TripScreen/KarhooTripPresenter.swift

```diff
@@ -185,7 +186,17 @@ final class KarhooTripPresenter: TripPresenter,
         cameraShouldFollowCar = true
 
         if trip.state == .driverEnRoute || trip.state == .arrived {
-            tripView?.focusOnUserLocation()
+            var isLocationPermissionGranted: Bool {
+                CLLocationManager.authorizationStatus() == .authorizedWhenInUse ||
+                    CLLocationManager.authorizationStatus() == .authorizedAlways
+            }
+            switch isLocationPermissionGranted {
+            case true:
+                tripView?.focusOnUserLocation()
+            case false:
+                tripView?.showNoLocationPermissionsPopUp()
+                tripView?.focusMapOnPickup()
```

Comment:
The map is focusing on the pickup location, but there is no pickup pin. There is a pin for the driver though, which is good. But we should show one for the pick-up too